IOPool/Streamer/src/EventMsgBuilder.cc

```diff
@@ -19,6 +20,13 @@ EventMsgBuilder::EventMsgBuilder(void* buf,
                                  uint32 adler_chksum,
                                  const char* host_name)
     : buf_((uint8*)buf), size_(size) {
+  uint32 expectedHeaderSize = computeHeaderSize(l1_bits.size(), hlt_bit_count);
+  if (expectedHeaderSize > size_) {
```

Comment:
This check should catch the cases like the one that triggered #37937, raising an exception with a (hopefully) readable error message instead of causing  a segmentation violation.